federation/pkg/kubefed/init/init.go

```diff
@@ -583,9 +587,51 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 		},
 	}
 
+	if dnsProviderConfig != "" {
+		// Create a dns-provider secret from the provided config file
+		dnsProviderConfigBytes, err := ioutil.ReadFile(dnsProviderConfig)
+		if err != nil {
+			return nil, err
+		}
+		secret := &api.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dnsProviderSecretName,
+				Namespace: namespace,
+			},
+			Data: map[string][]byte{
+				dnsProviderSecretKey: dnsProviderConfigBytes,
+			},
+		}
+		secret, err = clientset.Core().Secrets(namespace).Create(secret)
```

Comment:
(No action required) If an error occurs after the secret is created, the secret would linger.  I think this is a general problem with how ``kubefed`` operates.  Not saying it should be addressed in this PR, but I think it may make sense for ``kubefed`` to be able to clean up in the event of error (i.e. make ``init`` et al all-or-nothing).